### PR TITLE
various files: use explicit types in variable declarations

### PR DESCRIPTION
**Pull request by softworkz**

### CODE REVIEW

File: lib/ci-helper.ts [hunk 1]
```diff
@@ -754,7 +755,7 @@ export class CIHelper {
         if (result) {
             const results = await Promise.all(
                 commits.map((commit: IPRCommit) => {
-                const linter = new LintCommit(commit);
+                const linter: LintCommit = new LintCommit(commit);
```
Comment (webstech) [inline lib/ci-helper.ts:758]: Is this really needed?

File: lib/send-mail.ts [hunk 2]
```diff
@@ -117,7 +117,7 @@ export function parseMBoxMessageIDAndReferences(parsed: IParsedMBox):
         if (header.key === "In-Reply-To" || header.key === "References") {
             let value: string = header.value;
             while (value) {
-                const match = value.match(msgIdRegex);
+                const match: RegExpMatchArray | null = value.match(msgIdRegex);
```
Comment (webstech) [inline lib/send-mail.ts:120]: Is this really needed?

File: lib/ci-helper.ts (hunk 1 above)
Comment (softworkz) [reply lib/ci-helper.ts:758]: It shouldn't. Of course it should be inferred, don't know why it shows an issue.

File: lib/send-mail.ts (hunk 2 above)
Comment (softworkz) [reply lib/send-mail.ts:120]: same as above..

File: lib/ci-helper.ts (hunk 1 above)
Comment (webstech) [reply lib/ci-helper.ts:758]: What linter is saying there is an issue?

Comment (softworkz) [reply lib/ci-helper.ts:758]: Seems to be a ReSharper bug.

Without type spec:

image

With type spec:

image

File: lib/send-mail.ts (hunk 2 above)
Comment (softworkz) [reply lib/send-mail.ts]: I have updated the PR to remove those changes.

File: lib/ci-helper.ts (hunk 1 above)
Comment (softworkz) [reply lib/ci-helper.ts]: have updated the PR to remove those changes.

**Review by dscho — APPROVED** [review]
I am not necessarily convinced that those changes are _required_, but they don't hurt either, so I'll merge this PR.

**Comment by softworkz** [comment]
No they aren't logically required. But this solves the problem now while waiting for R# to fix it would take weeks or months :-)
Thank you!

**Pull request merged**